sdk/core/Azure.Core/src/Azure.Core.csproj

```diff
@@ -72,6 +72,11 @@
     <Compile Include="Shared\CancellationHelper.cs" />
     <Compile Include="Shared\OperationPoller.cs" />
     <Compile Include="Shared\TypeReferenceTypeAttribute.cs" />
+    <Compile Include="..\..\System.ClientModel\src\Internal\BitVector640.cs" />
```

Comment:
Yes, I agree -- and we have https://github.com/Azure/azure-sdk-for-net/issues/41035 tracking a holistic approach to sharing across Azure.Core and System.ClientModel -- this was just one more way I had implemented that sharing as a quick fix without actually duplicating the class file.

For this type, I do think it would only be used by the Core libraries today, but I know there's interest in using this type or a variant of it for the PATCH model work.  Something this general-purpose would probably need to be provided in a different `System.` namespace if we were to try to take it public, so we may need some copy/internal sharing approach while that effort bakes.  There's also a new C# feature that implements effective this -- it may be that we can use it instead if/when our repo upgrades to the right language version -- I can dig up the details on this as part of making this PR real.  

If nothing else, I like your idea to build an analyzer to enforce use of such a type by only the Core libraries -- I think we may have built something like that for `MutableJsonDocument` ... if I'm remembering correctly, we may be able to add an internal type to the list of types enforced by that analyzer and we would be good to go ... I can also dig up this information if we need it.

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-net/pull/42329